adapters/messaging-adapter-core/src/main/java/uk/gov/justice/services/adapter/messaging/JmsProcessor.java

```diff
@@ -30,15 +39,21 @@
      * @param message  a message to be processed
      */
     public void process(final Consumer<JsonEnvelope> consumer, final Message message) {
+
+        trace(LOGGER, () -> format("Processing message pre dispatching %s", toJmsTraceString(message)));
+
         if (!(message instanceof TextMessage)) {
             try {
-                throw new InvalildJmsMessageTypeException(String.format("Message is not an instance of TextMessage %s", message.getJMSMessageID()));
+                throw new InvalildJmsMessageTypeException(format("Message is not an instance of TextMessage %s", message.getJMSMessageID()));
             } catch (JMSException e) {
-                throw new InvalildJmsMessageTypeException(String.format("Message is not an instance of TextMessage. Failed to retrieve messageId %s",
+                throw new InvalildJmsMessageTypeException(format("Message is not an instance of TextMessage. Failed to retrieve messageId %s",
                         message), e);
             }
         }
 
-        consumer.accept(envelopeConverter.fromMessage((TextMessage) message));
+        final JsonEnvelope jsonEnvelope = envelopeConverter.fromMessage((TextMessage) message);trace(LOGGER, () -> format("Message converted to JsonEnvelope pre dispatching  %s", toEnvelopeTraceString(jsonEnvelope)));
+        consumer.accept(jsonEnvelope);
+        trace(LOGGER, () -> format("JsonEnvelope accepted by %s pre dispatching %s", consumer.getClass().toString(),
```

Comment:
Getting the consumer class isn't really meaningful, is it? It'll just be the dispatcher, or possibly, a lambda. I suggest we just log that the message has been processed